src/compiler/crystal/tools/playground/server.cr

```diff
@@ -470,7 +470,7 @@ module Crystal::Playground
 
       agent_ws = PathWebSocketHandler.new "/agent" do |ws, context|
         match_data = context.request.path.not_nil!.match(/\/(\d+)\/(\d+)$/).not_nil!
-        session_key = match_data[1]?.try(&.to_i)
+        session_key = match_data[1]?.try(&.to_i) || 0
```

Comment:
Why is this needed?

Reply:
`session_key` can't be `nil` because of [this line](https://github.com/crystal-lang/crystal/blob/2d5b06cf99172520ff733e05e8499f5ddafd94cc/src/compiler/crystal/tools/playground/server.cr#L477). Because it can't store `nil` and it's using `[]` instead of `[]?` below, this line seems like it _can't_ return nil even without this PR or it would raise a `KeyError` below.